src/tools.rs

```diff
@@ -361,6 +362,43 @@ pub async fn delete_files_in_dir(context: &Context, path: impl AsRef<Path>) -> R
     Ok(())
 }
 
+/// A guard which will remove the path when dropped.
+///
+/// It implements [`Deref`] so it can be used as a `&Path`.
+#[derive(Debug)]
+pub(crate) struct TempPathGuard {
+    path: PathBuf,
+}
+
+impl TempPathGuard {
+    pub(crate) fn new(path: PathBuf) -> Self {
+        Self { path }
+    }
+}
+
+impl Drop for TempPathGuard {
+    fn drop(&mut self) {
+        let path = self.path.clone();
+        tokio::spawn(async move {
```

Comment:
Another option is to remove files from async code (not from `drop()`), but move all the code that can fail to a separate function so that `remove_file()` can't be skipped easily. But probably all this isn't that important, exporting backups isn't a frequent operation